src/manager/utils.ts

```diff
@@ -251,7 +251,11 @@ export function shouldEnableEffect(
         logDebug(`Warning: wm_class_instance of ${win}: ${win.title} is null`);
         return false;
     }
-    if (getPref('blacklist').includes(wmClass)) {
+    if (!getPref('whitelist') && getPref('blacklist').includes(wmClass)) {
+        return false;
+    }
+    // Skip not whitelisted applications.
+    if (getPref('whitelist') && !getPref('blacklist').includes(wmClass)) {
         return false;
     }
```

Comment:
Can be simplified to

```ts
if (getPrefs('whitelist') != getPref('blacklist').includes(wmClass)) {
	return false;
}
```

Also, the schema keys should be changed to be less confusing, so this can be

```ts
const isException = getPref('blacklist').includes(wmClass);
const enableExceptions = getPrefs('whitelist_mode');
if (isException != enableExceptions) return false;
```